test/unittests/libethereum/BlockChain.cpp

```diff
@@ -209,7 +209,8 @@ BOOST_AUTO_TEST_CASE(Mining_5_BlockFutureTime)
     uncleBlock.mine(bc);
 
     BlockHeader uncleHeader = uncleBlock.blockHeader();
-    uncleHeader.setTimestamp(uncleHeader.timestamp() + 10000);
+    uncleHeader.setTimestamp(
+        dev::utcTime() + 1000000);  // mining does not reset timestamp on NoProof
```

Comment:
I don't understand this comment and what this change achieves.

Reply:
it makes uncleHeader timestamp in the future. 
with NoProof setTimestamp were not called. so this appeared to be in the past

Reply:
So should your `TestBlock::mine` or `TestBlock` constructor set the correct timestamp maybe?

Reply:
setting timestamp in the future is logic of this test case

Reply:
Ok, but `uncleHeader` should have the valid timestamp I think and then this change won't be needed

Reply:
need to investigate why this is failing. cause in calculateDifficulty PR it was fine